packages/gatsby-source-drupal/src/gatsby-node.js

```diff
@@ -264,4 +267,105 @@ exports.sourceNodes = async (
   for (const node of nodes) {
     createNode(node)
   }
+
+  if (process.env.NODE_ENV === `development` && preview) {
+    const server = micro(async (req, res) => {
```

Comment:
Could you elaborate why do we need to create separate server?

Reply:
There is no way to access the express instance in this method, if it were exposed I would use it

Reply:
Gotcha. I'll think about way to not have to do this. 

This introduces some potential issues - 8080 port is hardcoded, so there would be crashes when you have 2 drupal sources (or just trying to run 2 separate development servers with it). And instead of trying to address that it would be great to remove the need for spawning extra server

Reply:
Checking more on this - seems like thing that are referenced are `backRefs`/`addBackRef` (that doesn't allow easily moving it as-is). I do think we can address this. I'll try to do some work on this before weekend.

Reply:
I totally agree, I know there would be potential ussies with exposing it to all source plugins, I believe @DSchau mentioned something about dragons 🐲